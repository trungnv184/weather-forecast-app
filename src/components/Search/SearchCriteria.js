import React from "react";
import PropTypes from "prop-types";

const SearchCriteria = ({searchTerm}) => {
  return (
    <p className="title" data-testid="search-criteria">
      {searchTerm
        ? `Weather Forecast for location: ${searchTerm}`
        : "Please enter a location to get the weather forecast"}
    </p>
  );
};

SearchCriteria.propTypes = {
  searchTerm: PropTypes.string
};

export default SearchCriteria;
