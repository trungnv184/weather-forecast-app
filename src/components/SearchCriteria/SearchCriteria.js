import React from "react";
import PropTypes from "prop-types";

const SearchCriteria = ({searchTerm}) => {
  const getSearchCriteria = () => {
    return searchTerm
      ? `Weather Forecast for location: ${searchTerm}`
      : "Please enter a location to get the weather forecast";
  };

  return <p className="title">{getSearchCriteria()}</p>;
};

SearchCriteria.propTypes = {
  searchTerm: PropTypes.string
};

export default SearchCriteria;
