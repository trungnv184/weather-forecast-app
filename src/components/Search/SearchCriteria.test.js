import React from "react";
import {render, cleanup} from "@testing-library/react";
import {screen} from "@testing-library/dom";
import SearchCriteria from "./SearchCriteria";

afterEach(cleanup);

test("should display search criteria when weather api response data", () => {
  render(<SearchCriteria searchTerm="Ho Chi Minh City" />);
  expect(screen.getByTestId("search-criteria")).toHaveTextContent(
    "Weather Forecast for location: Ho Chi Minh City"
  );
});

test("should display message remind user input location to search", () => {
  render(<SearchCriteria />);
  expect(screen.getByTestId("search-criteria")).toHaveTextContent(
    "Please enter a location to get the weather forecast"
  );
});

test("snapshot", () => {
  const {container} = render(<SearchCriteria />);
  expect(container.firstChild).toMatchSnapshot();
});
