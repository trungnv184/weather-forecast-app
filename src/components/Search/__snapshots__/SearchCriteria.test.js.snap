// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<p
  class="title"
  data-testid="search-criteria"
>
  Please enter a location to get the weather forecast
</p>
`;
